### DISCUSSION TITLE
Is this architecture pratical ?

### DISCUSSION THREAD

**Discussion by intrastella** [community/community#46266] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Product Feedback

### Body

https://github.com/intrastella/dlPlotTool

I am a math student and hence have less experience with coding. Therefore, I would like to hear if parts of of this architecture is impractical.. ?

Thank you in advance :)

**Comment by airtower-luna** [comment, marked as answer]
I can't speak for the math/plotting logic, but from a Python developer and Git user perspective:

* `setup.py` is outdated, avoid it. You can use the [Setuptools declarative config](https://setuptools.pypa.io/en/latest/userguide/declarative_config.html) instead (in `setup.cfg`), or even better skip right to using [`pyproject.toml`](https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html) (which is the relatively new standard for Python packaging, not only Setuptools).
* Having an `__init__.py` at the top level of your repository seems odd at best. Usually you'll want to have a directory named after your package at the top level of the repository, and the Python files for that package inside it (tests and data files stay outside). The "Basic Use" section of the [Setuptools Quickstart guide](https://setuptools.pypa.io/en/latest/userguide/quickstart.html) has a simple example. This only really matters if you want to make an installable package, not just a collection of scripts, but I assume you do because otherwise having the `setup.py` file would be pointless.
* `pToolTest` looks a bit like you're writing your own test framework. You can do that, but it's much easier to use an existing one ([Pytest](https://docs.pytest.org/en/stable/) is very popular for good reasons, [`unittest`](https://docs.python.org/3.11/library/unittest.html) is a bit more complex to use but included in the standard library).
* There are a bunch of files that definitely or probably don't belong into the repository. In general, only source files should be in your repository, not generated ones. Ideally add the files you don't want to commit to `.gitignore`, so you don't add them by accident.
    * `__pycache__/`: Contains Python bytecode, which Python automatically generates (or updates) when running to make it faster to run the same code again with the same Python version.
    * There are a bunch of `std.log` files around the repository. I assume those are logs from running your code, not part of the code.
    * `.idea/`: People have different opinions on whether IDE-specific config should be part of the repository. Personally, I'm leaning towards "no".

I hope that helps!

**Reply by intrastella** [reply]
Thanks a lot! It's super helpful!
I used setup.py in that way bc I am used to that from my old work. All great insights. I will change that. :)
